### PR TITLE
Add pod annotations to RuntimeConfig

### LINKED ISSUE #7122: Support more CNI plugin capabilities
Is there any plan to support below CNI plugin capability?
`"capabilities": {
     "io.kubernetes.cri.pod-annotations": true
}`
This capability allows pass Pod annotations to CNI plugin.
Like (https://github.com/kubernetes/kubernetes/issues/69882#issuecomment-1509077177)

### PR DESCRIPTION
**Pull request by LenkaSeg**
#### Which issue(s) this PR fixes:
Fixes #7122 

#### Special notes for your reviewer:
Finishing touches of a PR authored by @welsper-young

#### Does this PR introduce a user-facing change?
See #7148 

```release-note
Add pod annotations to RuntimeConfig
```

### CODE REVIEW

**Review by saschagrunert — COMMENTED** [review]
Hey @LenkaSeg, thank you for the PR! I think the integration tests are failing now:

```
not ok 195 Ensure correct CNI plugin namespace/name/container-id arguments
```

File: test/cni_plugin_helper.bash [hunk 1]
```diff
@@ -60,3 +60,38 @@ EOF
...
+
+# this simple cni plugin just log stdin into file
+
+config=$(cat /dev/stdin)
+
+case $CNI_COMMAND in
+ADD)
+  log=$(jq -r '.config.log_path' <<<"$config")
+  if [[ "$log" != null ]]; then
+    echo "$config" >>"$log"
```
Comment (sohankunkerkar) [inline test/cni_plugin_helper.bash:72]: Try using a single `>` and see if that helps.

File: test/pod_annotations_capability.bats [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+@test "single cni plugin with pod annotations capability enabled" {
+	enable_capability=true
+	cni_plugin_log_path="$TESTDIR/cni.log"
+
+	start_crio
+	prepare_cni_plugin "$cni_plugin_log_path" $enable_capability
+	run_pod
+	delete_pod
+
+	annotations_cni_plugin_received=$(jq '.runtimeConfig."io.kubernetes.cri.pod-annotations"' "$cni_plugin_log_path")
```
Comment (haircommander) [inline test/pod_annotations_capability.bats:32]:
```
# jq: error: Could not open file /tmp/tmp.LCZOxyLqP3/cni.log: No such file or directory
```
this line fails in the tests still

Comment (sohankunkerkar) [reply test/pod_annotations_capability.bats:32]: Yeah, it's strange bc the same test passes locally.

File: test/helpers.bash [hunk 3]
```diff
@@ -596,3 +596,90 @@ function wait_until_exit() {
...
+        "log_path": "$3"
+      },
+      "capabilities": {
+        "io.kubernetes.cri.pod-annotations": $4
+      }
+    }
+  ]
+}
+EOF
+    chmod 755 "$TESTDATA"/"$CNI_PLUGIN_NAME".sh
```
Comment (haircommander) [inline test/helpers.bash:649]: can you try 777 to see if that helps the userns tests?

File: test/pod_annotations_capability.bats [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	crictl stopp "$id"
+	crictl rmp "$id"
+}
+
+@test "single cni plugin with pod annotations capability enabled" {
+	enable_capability=true
+	cni_plugin_log_path="$TESTDIR/cni.log"
+
+	start_crio
+	prepare_cni_plugin "$cni_plugin_log_path" $enable_capability
```
Comment (haircommander) [inline test/pod_annotations_capability.bats:28]: it's not clear to me, why are you ordering these this way? are you testing cri-o's reacting to the dropped in cni config or would it make sense to switch them?

File: server/sandbox_network.go [hunk 5]
```diff
@@ -230,6 +230,7 @@ func (s *Server) newPodNetwork(ctx context.Context, sb *sandbox.Sandbox) (ocicni
 	}
 
 	network := s.config.CNIPlugin().GetDefaultNetworkName()
+	podAnnotations := sb.Annotations()
```
Comment (MaoJianwei) [inline server/sandbox_network.go:233]: could you please make this change for 'newPodNetwork' and 'networkStop' functions? for cmdAdd and cmdDel.

thanks @LenkaSeg

**Review by MaoJianwei — CHANGES_REQUESTED** [review]
could you please make this change for 'newPodNetwork' and 'networkStop' functions? for cmdAdd and cmdDel.

thanks @LenkaSeg

**Comment by sohankunkerkar** [comment]
@cri-o/cri-o-maintainers  PTAL

**Comment by haircommander** [comment]
/approve

LGTM, @cri-o/cri-o-maintainers PTAL

**Review by sohankunkerkar — APPROVED** [review]
/lgtm

**Pull request merged**